### DISCUSSION TITLE
Markers have gone missing

### DISCUSSION THREAD

**Discussion by SlyChameleon** [javalent/obsidian-leaflet#129] [Q&A]
I'm hoping to get some help for an issue I've been having, and to find out if this is a known issue or if I should be posting this in "issues."

I wish I had more information, but the short version is that my map markers have gone missing.  They were there one day, and gone when I pulled the vault back up a few days later. The odd thing is that it appears marker data does exist for one of the 6 maps I had markers on, but they are markers for the wrong map (they are markers for a map that was moved to another vault).

As far as what might have caused the issue, I'm sure it's something I did, but I don't know what exactly may have caused the issue, so I'll post a few things about my setup and actions I took here, and maybe there is an obvious solution.

First, I use google drive to store my obsidian vaults, as I like to access them from two computers. The markers were working just fine between the two computers (once the plugin was installed on the second computer), and I have had no issues working from multiple computers for several weeks.

Second, my markers were pretty basic - they linked to headings in the same note that the map was in.

Third, I have reorganized folders a bit (via obsidian, not manually in file explorer) and I use the plugin in multiple vaults. I suspect shuffling files and reorganizing folders may have something to do with it, but I tried manually exploring the json files in the two vaults in which I use the leaflet plugin, and I cannot find the data (I'm assuming it would be found in the "data" json in the plugin folder).

That's pretty much it. I love the plugin, and will certainly make use of the export feature in the future, but if anyone knows how I might get my markers back, or how I could avoid having them wiped out, I would appreciate any help.

**Comment by valentine195** [comment]
Hmm, definitely not intended behavior. A few questions:

1. The plugin stores markers to maps by the map `id`. Did you happen to change these IDs in the code block?
2. If you go to settings and click "Export Marker Data", it will export the stored markers to a CSV file. If you do that, do you see the expected markers in there? (The format may be a little hard to read, sorry, but you should be able to find them via the link).

If they aren't in the exported data, they _probably_ aren't saved anymore. You could always go to the plugin folder (`Your Vault/.obsidian/plugins/obsidian-leaflet-plugin` and open the `data.json` file and look - that's where the plugin data is saved, so if its not there its not being read.

I will say that I pushed an update that could have accidentally deleted your markers if you opened a map, closed it, then re-opened it, and if that's the case I'm extremely sorry about that (it is actually what caused issue #123). If you have backups of your vault, the `data.json` file is probably also getting backed up and you should be able to recover them if you just copy it over. 

**Edit**: I should read your post fully before responding, I see you already talked about looking at the data file.

**Reply by SlyChameleon** [reply]
Thanks for the quick response!

I love your plugins btw. I'm using the dice roller in addition to the map one for D&D prep and session running, and it's working great (aside from the missing marker snafu).

I don't believe it was your recent update that did this, as I noticed the markers were missing before I updated the plugin (I updated the plugin hoping it would fix the issue).

I did rename one map, because I discovered early on (when making my second map), that the IDs needed to be different. I was getting all the map markers on both maps before I sorted that out. However, most of my maps have had unique IDs from the start, so I don't think that would be the issue.

I just tried exporting, but the marker data is not in there.

I do wish I knew what caused the problem, both to help potentially fix it and/or to help avoid it in the future. I suspect it has something to do with my re-arranging of the vaults I use though. Now that I think about it, I do think I might have done a copy/paste job (in windows explorer) to take folders from one vault and put them into a new separate folder for a new vault, and I might have accidentally moved the plugins folder, or the copy/paste job might have messed it up in some other way. I didn't delete anything that I'm aware of, but this may have messed some things up.

If I do run across the issue again, I'll let you know. Thanks to how easy the plugin is to use, it will not be too much trouble to re-create the markers, and I'll make occasional backups in the future, JUST in case :).

**Reply by valentine195** [reply]
Thanks. Unfortunately this was almost certainly on my end - the update that would have caused this was 3.17.0 (this specific commit: https://github.com/valentine195/obsidian-leaflet-plugin/commit/150a92a3d7fca75150d5bcfd26c93fb94e16dac0#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR483-R489)
